src/client/mod.rs

```diff
@@ -98,8 +98,8 @@ impl Client {
     ///
     /// See the documentation on
     /// [`ClientOptions::parse`](options/struct.ClientOptions.html#method.parse) for more details.
-    pub async fn with_uri_str(uri: &str) -> Result<Self> {
-        let options = ClientOptions::parse_uri(uri, None).await?;
+    pub async fn with_uri_str(uri: impl AsRef<str>) -> Result<Self> {
```

Comment:
I think we should also update the `collection`/`collection_with_options` and `database`/`database_with_options` methods to take `impl AsRef<str>` for the `name` params.

Reply:
Thank you for bringing this up, I meant to open a thread discussing this in my original review. 

Those two methods are a little tricky because they have generic parameters, so if we include `impl AsRef` as an argument, the turbofish operator will no longer work. If we introduce it as a regular generic argument instead, then users will have to specify both of them if they use the turbofish. When weighing all of these, I figured the inconsistency of requiring `&str` here was the best option of the three. What do you think? cc @kmahar

Reply:
sgtm